backends/python/server/text_embeddings_server/utils/device.py

```diff
@@ -37,7 +38,7 @@ def is_hpu() -> bool:
     is_hpu_available = True
     try:
         subprocess.run(["hl-smi"], capture_output=True, check=True)
-    except:
+    except subprocess.CalledProcessError:
```

Comment:
This change will cause crash on CPU/XPU, it will give error `FileNotFoundError: [Errno 2] No such file or directory: 'hl-smi'`